docs/src/writing-tests-java.md

```diff
@@ -0,0 +1,94 @@
+---
+id: writing-tests
+title: "Writing Tests"
+---
+
+Playwright assertions are created specifically for the dynamic web. Checks are automatically retried until the necessary conditions are met. Playwright comes with [auto-wait](./actionability.md) built in meaning it waits for elements to be actionable prior to performing actions. Playwright provides [assertThat](./test-assertions.md) overloads to write assertions.
+
+Take a look at the example test below to see how to write a test using web first assertions, locators and selectors.
+
+```java
+package org.example;
+
+import java.util.regex.Pattern;
+import com.microsoft.playwright.*;
+
+import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;
+
+public class App {
+    public static void main(String[] args) {
+        try (Playwright playwright = Playwright.create()) {
+            Browser browser = playwright.chromium().launch();
+            Page page = browser.newPage();
+            page.navigate("http://playwright.dev");
+
+            // Expect a title "to contain" a substring.
+            assertThat(page).hasTitle(Pattern.compile("Playwright"));
+
+            // create a locator
+            Locator getStarted = page.locator("text=Get Started");
+
+            // Expect an attribute "to be strictly equal" to the value.
+            assertThat(getStarted).hasAttribute("href", "/docs/intro");
+
+            // Click the get started link.
+            getStarted.click();
+
+            // Expects the URL to contain intro.
+            assertThat(page).hasURL(Pattern.compile(".*intro"));
+        }
+    }
+}
+```
+
+
+### Assertions
+
+Playwright provides [`assertThat`](./test-assertions.md) overloads which will wait until the expected condition is met.
+
+```java
+import java.util.regex.Pattern;
+import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;
+
+assertThat(page).hasTitle(Pattern.compile("Playwright"));
+```
+
+
+### Locators
+
+[Locators](./locators.md) are the central piece of Playwright's auto-waiting and retry-ability. Locators represent a way to find element(s) on the page at any moment and are used to perform actions on elements such as `.click` `.fill` etc. Custom locators can be created with the [`method: Page.locator`] method.
```

Comment:
This page needs an overview/intro section that will mention key concepts like web assertions, locators and how they are related. Reading the doc in its current form I cannot make a connection between assertions and locators.

Reply:
we have it in all the langs, lets follow up with it.